molecule/provisioner/ansible.py

```diff
@@ -407,6 +407,10 @@ def default_options(self):
 
     @property
     def default_env(self):
+        # Allow user to use ~/path/to/foo in environment variables
+        for key, value in self._config.env.items():
+            self._config.env[key] = os.path.expanduser(value)
+
```

Comment:
This worries me a little bit as it would break if genuine `~` is present in other environment variables, ones that do not need/want to be expanded. I am not aware of any existing such cases but I can imagine some could.

Another issue that I see even more problematic is that we have ~20 implementations of `default_env()` across the code base and that is addressing only in one place.

I do not see any reason why this expansion should happen only for provisioner/ansible and not for others. While I can ignore lack of tests for this one I am afraid that the second one is more serious.... also a reason to refactor the code to avoid that duplication and have a single implementation.

Reply:
> This worries me a little bit as it would break if genuine ~ is present in other environment variables, ones that do not need/want to be expanded.

Which vars do you have that have a `~` in them? We could just add a global `--no-expand` if needed. We should test around this to satisfy your concern.


> Another issue that I see even more problematic is that we have ~20 implementations of default_env() across the code base and that is addressing only in one place.

> I do not see any reason why this expansion should happen only for provisioner/ansible and not for others. While I can ignore lack of tests for this one I am afraid that the second one is more serious.... also a reason to refactor the code to avoid that duplication and have a single implementation.

Those are problems but let's not push this on a contributor who is trying to support their case and not deal with all the things wrong with molecule and variable expansion.